numpy/lib/arrayterator.py

```diff
@@ -124,14 +124,17 @@ def __getitem__(self, index):
             out.stop[i] = min(stop, out.stop[i])
         return out
 
-    def __array__(self):
+    def __array__(self, dtype=None):
```

Comment:
Looks to me like `dtype` should be a positional argument:
```
>>> x = np.arange(5)
>>> x.__array__(np.float32)
array([0., 1., 2., 3., 4.], dtype=float32)
>>> x.__array__(dtype=np.float32)
Traceback (most recent call last):
  File "<ipython-input-43-6adb0d04469e>", line 1, in <module>
    x.__array__(dtype=np.float32)
TypeError: __array__() takes no keyword arguments
```

Reply:
Based on some trial and error it seems that the `dtype` parameter must have a default value but can otherwise be either position-only or positional-or-keyword (`ndarray` being an example of the former).

Secondly, I'm not convinced anymore that this change is even necessary in the first place,
as I was initially under the (incorrect) impression that a `dtype` parameter would be required here 
if one wants to use it with the likes of, _e.g._, `np.array(x, dtype=<insert dtype different from x's dtype>`).

I think it might be best to revert this change as well and then make the `npt._SupportsArray` protocol a bit more flexible, such that it can accomodate `__array__` without the `dtype` parameter.

Reply:
Yes, adding the `dtype` keyword is not needed, reverting sounds good.